modules/dnn/src/int8layers/quantization_utils.cpp

```diff
@@ -35,43 +38,123 @@ static void broadcast1D2TargetMat(Mat& data, const MatShape& targetShape, int ax
+static void block_repeat(InputArray src, int axis, int repetitions, OutputArray dst)
 {
-    // broad cast the scales and zeropoint to the input shape.
-    MatShape subTargetShape(targetShape.size(), 1);
-    subTargetShape[axis] = scales.size();
+    CV_Assert(src.getObj() != dst.getObj());
+    CV_Check(axis, axis >= 0 && axis < src.dims(), "Axis out of range");
+    CV_CheckGT(repetitions, 1, "More than one repetition expected");
+
+    Mat src_mat = src.getMat();
+    Mat dst_mat;
+
+    MatShape sshape = shape(src_mat);
+    MatShape dshape = sshape;
+
+    size_t dtype_bytes = src_mat.elemSize();
+    int chunk_size = dtype_bytes;
+    int num_chunks = 1;
+
+    dshape[axis] *= repetitions;
+
+    for (int i = axis+1; i < src.dims(); ++i)
+        chunk_size*=sshape[i];
 
-    zeropointsMat.create(subTargetShape.size(), subTargetShape.data(), CV_32FC1);
-    scalesMat.create(subTargetShape.size(), subTargetShape.data(), CV_32FC1);
+    for (int i = 0; i <= axis; ++i)
+        num_chunks*=sshape[i];
 
-    const int len = scales.size();
-    // Deep copy the scales and zeropoint data and prevent the original data from being changed.
+    dst.create(src.dims(), dshape.data(), src.type());
+    dst_mat = dst.getMat();
+
+
+    for (int i = 0; i < repetitions; ++i) {
+        size_t src_offset = 0;
+        size_t dst_offset = i * chunk_size;
+
+        for (int j = 0; j < num_chunks; ++j) {
+            memcpy(dst_mat.data + dst_offset, src_mat.data + src_offset, chunk_size);
+            src_offset += chunk_size;
+            dst_offset += chunk_size * repetitions;
+        }
+    }
+}
+
+template <typename T>
+static void copyVecToMat(Mat& mat, const std::vector<T>& data){
+    float * matPtr = mat.ptr<float>(0);
+    const int len = data.size();
 
-    float * scalePtr = scalesMat.ptr<float>(0);
     for (int i = 0; i < len; i++)
-        scalePtr[i] = scales[i];
+        matPtr[i] = (float) data[i];
+}
 
-    float * zpPtr = zeropointsMat.ptr<float>(0);
+template <typename T>
+static void copyMatToVec(const Mat& mat, std::vector<T>& data){
+    const float * matPtr = mat.ptr<float>(0);
+    const int len = mat.total();
+    data.clear();
     for (int i = 0; i < len; i++)
-        zpPtr[i] = (float )zeropoints[i];
+        data.push_back(matPtr[i]);
+}
 
-    broadcast1D2TargetMat(scalesMat, targetShape, axis);
-    broadcast1D2TargetMat(zeropointsMat, targetShape, axis);
+template <typename T>
+static void broadcastBlockedMatrix(Mat& mat, const std::vector<T>& data, const MatShape& targetShape, int axis, int block_size){
+    CV_Check(block_size, targetShape[axis] % block_size == 0 && block_size <= targetShape[axis], "Block size must be a divisor of the target dimension size and not exceed it.");
+
+    MatShape subTargetShape(targetShape);
+    subTargetShape[axis] = static_cast<int>(subTargetShape[axis] / block_size);
+    Mat tmpMat(subTargetShape.size(), subTargetShape.data(), CV_32FC1);
+
+    copyVecToMat(tmpMat,data);
```

Comment:
Why do you need a copy here? Also you can push std::vector<> as input to function with InputArray. It's handled automatically.

Reply:
Do you suggest to specialize `block_repeat` to handle std::vector so we can avoid a copy?

Reply:
Done! Thank you for the suggestion.